src/components/MainContent.css

```diff
@@ -0,0 +1,20 @@
+.input:not(:placeholder-shown) + .label {
```

Comment:
This kind of label and input treatment is something you might often want to do. No changes immediately needed here, but one thing you can think about is putting all of this on a more configuration level (use utilities and tailwind's extend, either make classes or apply them in the tailwind's way).

You might imagine one day on another page, where you want to apply the same thing, and you'd be reaching inside MainContent.css (from how you have it set up). Instead, you can set that up on tailwind's overall config level and have it accessible anywhere (just like the default classes they come with).